frappe/email/doctype/email_queue/email_queue.js

```diff
@@ -4,12 +4,13 @@
 frappe.ui.form.on("Email Queue", {
 	refresh: function(frm) {
 		if (["Not Sent","Partially Sent"].indexOf(frm.doc.status)!=-1) {
-			frm.add_custom_button("Send Now", function() {
+			let button = frm.add_custom_button("Send Now", function() {
 				frappe.call({
 					method: 'frappe.email.doctype.email_queue.email_queue.send_now',
 					args: {
 						name: frm.doc.name
 					},
+					btn: button,
```

Comment:
Just a question: Variables declared using `let` are not not initialised until their definition is evaluated. So why does using `button` while initialising it work?

:exploding_head:

Reply:
Because the callback function in which `button` is used is not evaluated yet. It will be evaluated when button is clicked, and by then `button` is initialised.

Reply:
I tried following code using similar logic:
```
let x = () => console.log(x);
```
And it worked! Closures I guess. :exploding_head:

Reply:
Closure + lazy evaluation